### PR TITLE
Split up the Scala example in parts, #199

### PR DESCRIPTION
**Pull request by patriknw**
No point in reviewing full PR. Look at the "modified" commit, and the `copy-identical-files.sh` script.

References #199

### CODE REVIEW

File: copyIdenticalFiles.sh [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+# from seed
+declare SRC="${tutorial_root}/00-shopping-cart-service-scala/build.sbt"
+cp ${SRC} ${tutorial_root}/01-shopping-cart-service-scala/
+cp ${SRC} ${tutorial_root}/02-shopping-cart-service-scala/
+cp ${SRC} ${tutorial_root}/03-shopping-cart-service-scala/
+cp ${SRC} ${tutorial_root}/04-shopping-cart-service-scala/
+cp ${SRC} ${tutorial_root}/05-shopping-cart-service-scala/
+cp ${SRC} ${tutorial_root}/shopping-cart-service-scala/
+cp ${SRC} ${howto_root}/shopping-cart-service-scala/
```
Comment (ignasi35) [inline copyIdenticalFiles.sh:14]: I think we can't use this last one. The copy of `build.sbt` in `${howto_root}/shopping-cart-service-scala/` has a handful of edits now (introduced in #226).

TBH, I'd rather undo the change in #226 and keep `${howto_root}/shopping-cart-service-scala/` in the list of automatically-maintained files.

File: copyIdenticalFiles.sh [hunk 2]
```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env bash
```
Comment (ignasi35) [inline copyIdenticalFiles.sh:1]: @patriknw I've only reviewed this file assuming all the rest are generated (copied) files.

**Review by ignasi35 — COMMENTED** [review]
Small note regarding a collision with #226

File: copyIdenticalFiles.sh (hunk 2 above)
Comment (patriknw) [reply copyIdenticalFiles.sh:1]: good, you can look at the commits named "modified..." but I can complete the full thing first if you think this is on the right track

**Comment by ignasi35** [comment]
@patriknw what's your plan for `copyIdenticalFiles.sh`? Should we have a job in Travis that runs ths script and fails the job if `git diff` detects changes?

**Comment by patriknw** [comment]
> Should we have a job in Travis that runs ths script and fails the job if git diff detects changes?

Yes, I was thinking that could be a simple way to make sure things are in sync.

File: copyIdenticalFiles.sh (hunk 2 above)
Comment (ignasi35) [inline copyIdenticalFiles.sh:1]: We have a `scripts/` in the root pf the repo. Maybe we could move this script there.

Comment (patriknw) [reply copyIdenticalFiles.sh]: done

**Comment by patriknw** [comment]
@ignasi35 @octonato I have completed this. Ready for review.

File: .travis.yml [hunk 3]
```diff
@@ -45,6 +66,10 @@ jobs:
         CMD="test; scalafmtCheckAll"
       name: "Howto cleanup dependencies"
 
+    script:
+      - scripts/copy-identical-files.sh; git diff --exit-code --color || { echo "[error] Found modified file that was expected to be identical. Run scripts/copy-identical-files.sh"; false; }
```
Comment (ignasi35) [inline .travis.yml:70]: nice!

**Review by ignasi35 — COMMENTED** [review]
So, IIUC, the `00-shopping-cart` is the source for all the common files.

I just noticed the `00-shopping-cart/src/main/scala/.../Main.scala` exists but is not a source for any duplicate (it is not used from the copy-identical-files.sh scripts. That's probably a left-over, or is it missing from the copy-identical-files(?).

What is the plan to maintain the files in `00-shopping-cart` in sync with the microservices g8 template then? Maybe we can add a small list of steps for the planned maintenance.

**Comment by patriknw** [comment]
> I just noticed the 00-shopping-cart/src/main/scala/.../Main.scala exists but is not a source for any duplicate (it is not used from the copy-identical-files.sh scripts. That's probably a left-over, or is it missing from the copy-identical-files(?).

00 is exactly what the g8 template generates. The Main there is probably not used elsewhere because it doesn't start anything.

> What is the plan to maintain the files in 00-shopping-cart in sync with the microservices g8 template then? Maybe we can add a small list of steps for the planned maintenance.

We could automatically pull in latest g8 and update the 00 sources automatically, but I don't want us to invest in infra for g8 until we know what templating we will use for real. For now it's manual update.
Let me write something in contributing.

**Comment by ignasi35** [comment]
> […] but I don't want us to invest in infra for g8 until we know what templating we will use for real. For now it's manual update.

Hell no. Manual is great.

> Let me write something in contributing.

:-)

**Comment by patriknw** [comment]
@ignasi35 I have added CONTRIBUTING.md in https://github.com/akka/akka-platform-guide/pull/225/commits/e6b4e4b781586cb5c16dd2cf4723222426742789

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+Contributions to the documentation are welcome and encouraged.
+If you are unfamiliar with the project or with asciidoc, please read the contribution guidelines below.
+
+## Contributing to the Akka Documentation
+
+Detailed information about working with the documentation is provided in the [docs-source](docs-source/README.adoc) folder.
+
+## Building the Documentation
+
+This part of the Akka documentation is built using [Antora](https://docs.antora.org/antora/2.1/), from asciidoc sources.
```
Comment (ignasi35) [inline CONTRIBUTING.md:30]:
```suggestion
This part of the Akka documentation is built using [Antora](https://docs.antora.org/antora/2.3/), from asciidoc sources.
```

😅

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    List all the TODOs, review comments, unresolve references, etc. from the documentation.
+
+# Synchronize code for different parts of the tutorial
+
+Each part of the tutorial has full source code in directories in `docs-source/docs/modules/microservices-tutorial/examples/`.
+
+Many files are identical and maintained with `scripts/copy-identical-files.sh`, which copies the files.
+
+The `00` directory corresponds to the generated code from https://github.com/akka/akka-microservices-seed-scala.g8.
+When changes are made to the g8 temaplate the `00` directory should be updated. It can be created with `sbt new`
```
Comment (ignasi35) [inline CONTRIBUTING.md:63]:
```suggestion
When changes are made to the g8 template the `00` directory should be updated. It can be created with `sbt new`
```

File: README.md [hunk 6]
```diff
@@ -30,52 +25,15 @@ The example application used in the documentation is composed of 3 three microse
 * [Shopping cart analytics service in Scala](docs-source/docs/modules/microservices-tutorial/examples/shopping-analytics-service-scala)
 * [Order service in Scala](docs-source/docs/modules/microservices-tutorial/examples/shopping-order-service-scala)
 
+Contributing
+------------
+Contributions are *very* welcome!
 
-Antora-based Akka Documentation
--------------------------------
```
Comment (ignasi35) [inline README.md:35]: Good to keep the README cleaner.

**Review by ignasi35 — APPROVED** [review]
minor typos

**Comment by ignasi35** [comment]
I squashed the commits by mistake. Sorry  :-(
